pysys/baserunner.py

```diff
@@ -228,23 +243,45 @@ def testComplete(self, testObj, dir):
 			removeNonZero = False
 
 		try:
-			for file in os.listdir(toLongPathSafe(dir)):
-				path = toLongPathSafe("%s/%s" % (dir, file), onlyIfNeeded=True)
-				if PLATFORM in ['sunos', 'linux']:
-					size = os.lstat(path)[stat.ST_SIZE]
-				else:
-					size = os.stat(path)[stat.ST_SIZE]
-
-				if (size == 0) or (removeNonZero and 'run.log' not in file and self.isPurgableFile(path)):
-					count = 0
-					while count < 3:
-						try:
-							os.remove(path)
-							break
-						except Exception:
-							if not os.path.exists(path): break
-							time.sleep(0.1)
-							count = count + 1
+			for (dirpath, dirnames, filenames) in os.walk(toLongPathSafe(dir), topdown=False):
+				deleted = 0
+				for file in filenames:
+					path = os.path.join(dirpath, file)
+					for collect in self.__collectTestOutput:
+						if fnmatch.fnmatch(os.path.basename(file), collect['pattern']):
+							collectdest = os.path.join(mkdir(collect['outputDir']), (collect['outputPattern']
+								.replace('@TESTID@', str(testObj))
```

Comment:
Worth explicitly stating somewhere that TESTID, FILENAME, UNIQUEID are the _only_ possible substitutions? It's implied but not actually confirmed until you read the code.

Reply:
yes good point I'll add that detail to the USERGUIDE.rst